internal/docker_test.go

```diff
@@ -0,0 +1,480 @@
+	remainingCntr := []types.Container{}
+
+	for _, cntr := range cntrList {
+		if cntr.ID != containerID {
+			remainingCntr = append(remainingCntr, cntr)
+		}
+	}
+
+	return remainingCntr
+}
+
+// getContainerStats returns the container's stats given it's id and a list of all the containers.
+func getContainerStats(cntrList []types.Container, containerID string) (types.ContainerStats, error) {
+	found := 0
+
+	for _, cntr := range cntrList {
+		if cntr.ID == containerID {
+			found = 1
+
+			break
+		}
+	}
+
+	if found == 0 {
+		return types.ContainerStats{}, &ContainerError{Type: ErrContainerNotFound}
+	}
+
+	readerVar := ioutil.NopCloser(bytes.NewReader([]byte(readerCloserStats)))
+	stats := types.ContainerStats{Body: readerVar, OSType: "linux"}
+
+	return stats, nil
+}
+
+func TestDockerFuncs(t *testing.T) {
+	Convey("Test to check nanosecondsToSec", t, func() {
+		So(nanosecondsToSec(634736438394834), ShouldEqual, 634736)
+		So(nanosecondsToSec(634736), ShouldEqual, 0)
+		So(nanosecondsToSec(0), ShouldEqual, 0)
+	})
+
+	Convey("Test to check bytesToMB", t, func() {
+		So(bytesToMB(634736438), ShouldEqual, 605)
+		So(bytesToMB(1048576), ShouldEqual, 1)
+		So(bytesToMB(0), ShouldEqual, 0)
+	})
+
+	Convey("Given a NewDockerClient get the current container list", t, func() {
+		// Create a list of dummy containers
+		cntrList := []types.Container{{
+			ID: "container_id1", Names: []string{"/test_container1"},
+		}, {
+			ID: "container_id2", Names: []string{"/test_container2"},
+		}, {
+			ID: "container_id3", Names: []string{"/test_container3"},
+		}}
+
+		// Create a docker client with list of dummy containers
+		dockerClient := NewDockerClient(
+			&mockDockerClient{
+				ContainerListFn: func() ([]types.Container, error) {
+					return cntrList, nil
+				},
+
+				ContainerKillFn: func(containerID string) error {
+					remainingCntr := removeContainer(cntrList, containerID)
+					if len(cntrList) == len(remainingCntr) {
+						return &ContainerError{Type: ErrContainerNotFound}
+					}
+
+					// Copy the remaining containers to cntrList
+					cntrList = remainingCntr
+
+					return nil
+				},
+
+				ContainerStatsFn: func(containerID string) (types.ContainerStats, error) {
+					return getContainerStats(cntrList, containerID)
+				},
+			},
+		)
+
+		clist, err := dockerClient.GetCurrentContainers()
+		So(err, ShouldBeNil)
+		So(clist, ShouldNotBeNil)
+		So(len(clist), ShouldEqual, 3)
+
+		// Create a docker client with no containers
+		empDockerClient := NewDockerClient(&mockDockerClient{ContainerListFn: func() ([]types.Container, error) {
+			return []types.Container{}, &ContainerError{Type: ErrContainerListEmpty}
+		}})
+
+		emplist, err := empDockerClient.GetCurrentContainers()
+		So(err, ShouldNotBeNil)
+		So(emplist, ShouldBeEmpty)
+		So(len(emplist), ShouldEqual, 0)
+
+		// Create some files containing container id
+		dockerTempDir, err := ioutil.TempDir("/tmp", "docker_temp_")
```

Comment:
Don't specify `/tmp`, just leave it blank so the user's desired tmp location will be used

Reply:
Updated